pkg/operator/controller/status_test.go

```diff
@@ -434,3 +437,303 @@ func TestOperatorStatusesEqual(t *testing.T) {
 		}
 	}
 }
+
+func TestComputeOperatorStatusVersions(t *testing.T) {
+	type testInputs struct {
+		oldReleaseVersion, oldIngressControllerImage string
+		curReleaseVersion, curIngressControllerImage string
+		availableCondition, progressingCondition     bool
+	}
+	type testOutputs struct {
+		releaseVersion, ingressControllerImage string
+		versionChanged, ingressImageChanged    bool
+	}
+	testCases := []struct {
+		description string
+		inputs      testInputs
+		outputs     testOutputs
+	}{
+		{
+			description: "initialize versions, operator is not available and not progressing",
+			inputs: testInputs{
+				oldReleaseVersion:         UnknownVersionValue,
+				oldIngressControllerImage: UnknownVersionValue,
+				curReleaseVersion:         "v1",
+				curIngressControllerImage: "ic-v1",
+			},
+			outputs: testOutputs{
+				releaseVersion:         UnknownVersionValue,
+				ingressControllerImage: UnknownVersionValue,
+				versionChanged:         true,
+				ingressImageChanged:    true,
+			},
+		},
+		{
+			description: "initialize versions, operator is not available and progressing",
+			inputs: testInputs{
+				oldReleaseVersion:         UnknownVersionValue,
+				oldIngressControllerImage: UnknownVersionValue,
+				curReleaseVersion:         "v1",
+				curIngressControllerImage: "ic-v1",
+				progressingCondition:      true,
+			},
+			outputs: testOutputs{
+				releaseVersion:         UnknownVersionValue,
+				ingressControllerImage: UnknownVersionValue,
+				versionChanged:         true,
+				ingressImageChanged:    true,
+			},
+		},
+		{
+			description: "update with no change",
+			inputs: testInputs{
+				oldReleaseVersion:         "v1",
+				oldIngressControllerImage: "ic-v1",
+				curReleaseVersion:         "v1",
+				curIngressControllerImage: "ic-v1",
+				availableCondition:        true,
+			},
+			outputs: testOutputs{
+				releaseVersion:         "v1",
+				ingressControllerImage: "ic-v1",
+			},
+		},
+		{
+			description: "update operator version, operator is not available and not progressing",
+			inputs: testInputs{
+				oldReleaseVersion:         "v1",
+				oldIngressControllerImage: "ic-v1",
+				curReleaseVersion:         "v2",
+				curIngressControllerImage: "ic-v1",
+			},
+			outputs: testOutputs{
+				releaseVersion:         "v1",
+				ingressControllerImage: "ic-v1",
+				versionChanged:         true,
+			},
+		},
+		{
+			description: "update operator version, operator is available and progressing",
+			inputs: testInputs{
+				oldReleaseVersion:         "v1",
+				oldIngressControllerImage: "ic-v1",
+				curReleaseVersion:         "v2",
+				curIngressControllerImage: "ic-v1",
+				progressingCondition:      true,
+				availableCondition:        true,
+			},
+			outputs: testOutputs{
+				releaseVersion:         "v1",
+				ingressControllerImage: "ic-v1",
+				versionChanged:         true,
```

Comment:
Sam as my feedback above. It seems like `versionChanged` should be `false` if the outputs versions do not change.